src/blog/on-removing-let-let-mut.vmd

```diff
@@ -0,0 +1,330 @@
+    * In C++, `auto x = 4;`
+    * Swift's `var x = 4`
+    * Go's `x := 4` (though it's not really a keyword per se)
+ * Specifying whether the variable can change, such as:
+    * Rust's `let x = 4;` vs `let mut x = 4;`
+    * Javascript's `let x = 4` vs `const x = 4`
+ * Leaving it off completely, such as:
+    * Python's `x = 4`
+
+
+The previous version of [Vale](https://vale.dev/) used the second option, `let` and `let mut`. [# Vale is a language that aims to be fast and memory-safe, while still being easy to learn. This syntax change helps with that!]
+
+However, it recently changed to a fourth option, which was so nice that it became the default behavior in version 0.2.
+
+
+<slice/>
+
+
+# The Fourth Option
+
+Most languages can't just use `x = 4;`, because that already meant something; that's the *re-assignment statement*, which modifies an existing variable that we already declared.
+
+And alas, Python tried combining those two statements. It didn't go well. [# This leads to problems; if you rename your `x = 1;` declaration to `y = 1;`, but forget to modify `x = 42;` re-assignment below, you now accidentally have two variables!]
+
+
+However, there's another option here: *let's change the re-assignment statement!*
+
+
+<<<<
+Instead of this:
+
+```vale
+func main() {
+  let a = 3;
+  let b = 3;
+  let c = 3;
+  let mut d = 3;
+  d = 7;
+  println(d);
+}
+```stdout
+7
+```
+////
+We can have this:
+
+```vale
+func main() {
+  a = 3;
+  b = 3;
+  c = 3;
+  d = 3;
+  set d = 7;
+  println(d);
+}
+```stdout
+7
+```
+>>>>
+
+In other words, `x = 3;` declares a variable, and `set x = 42;` re-assigns it.
+
+
+It was odd at first, but after using it for a few weeks, we think this is a huge improvement.
+
+
+Syntax design can be a tricky endeavor. When exploring new syntax, we had to suppress the knee-jerk "unfamiliarity avoidance", and actually experiment. Without experimenting, it's easy to get stuck with what's familiar, even if there are better options.
+
+
+<slice/>
+
+
+# Why We Like It
+
+To our great surprise, we've found that *our codebases have a lot more declarations than re-assignments,* so it makes sense to require the extra keyword on re-assignments because they're rarer.
+
+
+We sampled three Vale projects. One had 111 declarations, and only 35 re-assignments. That's only *21%* re-assignments! The other two were even lower, at *20%* and *6%*.
+
+This isn't just Vale either. A randomly chosen Rust library, [Rocket](https://github.com/SergioBenitez/Rocket), had about *8%*. [#rocket]
+
+
+*When did this happen?* We used to re-assign variables all the time!
+
+<slice>
+#rocket: This is approximate; I used `let(\s+mut)?` (4437 results) and `^\s*[\w\[\]\.]+\s*[\+\-\*\/]?=\s+` (351 results) and which may have missed some corners, such as lambdas and mutating call returned values.
+
+Parsing with regular expressions is fun!
+</slice>
+
+
+We suspect three things contributed to this shift over the past couple decades.
```

Comment:
I would recommend staying general here, as I think there are more than 3 things that have changed this. Variable shadowing becoming more popular is another feature I would say has contributed to the shift.
Instead "We suspect moving towards more declarative patterns has contributed to this shift... .  Such as these three things"